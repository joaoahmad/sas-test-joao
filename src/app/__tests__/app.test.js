import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom';
import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../root-reducer';
import App from '../app';

const mockStore = (state = {}) => createStore(rootReducer, state, applyMiddleware(thunk));

describe('A <CallDetails> component', () => {
  let wrapper;

  it('should render correctly', () => {
    const store = mockStore();

    wrapper = mount((
      <Provider store={store}>
        <MemoryRouter initialEntries={['/']} initialIndex={0}>
          <App />
        </MemoryRouter>
      </Provider>
    ));

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render logged correctly', () => {
    const store = mockStore({
      operator: {
        operator_name: 'Test Name',
      }
    });

    wrapper = mount((
      <Provider store={store}>
        <MemoryRouter initialEntries={['/']} initialIndex={0}>
          <App />
        </MemoryRouter>
      </Provider>
    ));

    expect(wrapper.html()).toMatchSnapshot();
  });
});
