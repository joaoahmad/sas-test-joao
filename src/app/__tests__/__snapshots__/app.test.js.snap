// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A <CallDetails> component should render correctly 1`] = `"<div><div class=\\"app\\"><div class=\\"operator-login\\"><div class=\\"operator-login-modal\\"><form><h1>Entrar</h1><p>Digite seu nome para continuar</p><input name=\\"operator_name\\" placeholder=\\"Seu nome\\" value=\\"\\"><button type=\\"submit\\" class=\\"button\\">Entrar</button></form></div></div><div class=\\"app-navigation\\"><div class=\\"navigation\\"></div></div><div class=\\"app-list\\"><a type=\\"link\\" color=\\"white\\" class=\\"button button--white button-new\\" href=\\"/calls/new\\">Add Chamado</a><div>loading...</div></div><div class=\\"app-main\\"><div class=\\"new-call\\"><h1>Novo Chamado</h1><form><div class=\\"form-row\\"><label>Atendente</label><input name=\\"operator\\" type=\\"text\\" disabled=\\"\\" value=\\"\\"></div><div class=\\"form-row\\"><label>Motivo do Chamado</label><select name=\\"reason\\"><option>Dúvidas</option><option>Elogios</option><option>Sugestões</option><option>Reclamações</option><option>Outros</option></select></div><div class=\\"form-row\\"><label>Tipo de Chamado</label><select name=\\"type\\"><option>Telefone</option><option>Chat</option><option>Email</option></select></div><div class=\\"form-row\\"><label>Estado</label><select name=\\"state\\"><option>AC</option><option>AL</option><option>AM</option><option>AP</option><option>BA</option><option>CE</option><option>DF</option><option>ES</option><option>GO</option><option>MA</option><option>MG</option><option>MS</option><option>MT</option><option>PA</option><option>PB</option><option>PE</option><option>PI</option><option>PR</option><option>RJ</option><option>RN</option><option>RO</option><option>RR</option><option>RS</option><option>SC</option><option>SE</option><option>SP</option><option>TO</option></select></div><div class=\\"form-row\\"><label>Detalhes</label><textarea name=\\"description\\"></textarea></div><button type=\\"submit\\" class=\\"button\\">Enviar</button></form></div></div></div></div>"`;

exports[`A <CallDetails> component should render logged correctly 1`] = `"<div><div class=\\"app\\"><div class=\\"app-navigation\\"><div class=\\"navigation\\"></div></div><div class=\\"app-list\\"><a type=\\"link\\" color=\\"white\\" class=\\"button button--white button-new\\" href=\\"/calls/new\\">Add Chamado</a><div>loading...</div></div><div class=\\"app-main\\"><div class=\\"new-call\\"><h1>Novo Chamado</h1><form><div class=\\"form-row\\"><label>Atendente</label><input name=\\"operator\\" type=\\"text\\" disabled=\\"\\" value=\\"Test Name\\"></div><div class=\\"form-row\\"><label>Motivo do Chamado</label><select name=\\"reason\\"><option>Dúvidas</option><option>Elogios</option><option>Sugestões</option><option>Reclamações</option><option>Outros</option></select></div><div class=\\"form-row\\"><label>Tipo de Chamado</label><select name=\\"type\\"><option>Telefone</option><option>Chat</option><option>Email</option></select></div><div class=\\"form-row\\"><label>Estado</label><select name=\\"state\\"><option>AC</option><option>AL</option><option>AM</option><option>AP</option><option>BA</option><option>CE</option><option>DF</option><option>ES</option><option>GO</option><option>MA</option><option>MG</option><option>MS</option><option>MT</option><option>PA</option><option>PB</option><option>PE</option><option>PI</option><option>PR</option><option>RJ</option><option>RN</option><option>RO</option><option>RR</option><option>RS</option><option>SC</option><option>SE</option><option>SP</option><option>TO</option></select></div><div class=\\"form-row\\"><label>Detalhes</label><textarea name=\\"description\\"></textarea></div><button type=\\"submit\\" class=\\"button\\">Enviar</button></form></div></div></div></div>"`;
